homework-1/task_1.py

```diff
@@ -0,0 +1,75 @@
+"""
+
+TASK 1
+
+Write a class to represent a Cash Register.
+You class should keep the state of price total and purchased items
+
+Below is a starter code:
+------------------------
+1. you can add new variables and function if you want to
+2. you will NEED to add accepted method parameters where required.
+For example, method add_item probably accepts some kind of an item?..
+3. You will need to write some examples of how your code works.
+
+"""
+
+
+class CashRegister:
+
+    def __init__(self):
+
+        self.total_items = []
+        self.total_price = 0
+        self.discount = 0
+
+    def add_item(self, item):
+        self.total_items.append(item)
+        print(f"You have added {item} to the shopping basket.")
+
+    def remove_item(self, item):
+        self.total_items.remove(item)
+        print(f"You have removed {item} from the shopping basket.")
+
+    def apply_discount(self):
+        """Apply a discount, buy one get one free."""
+        unique_items = []
+        cost_of_unique_items = 0
+        for item in self.total_items:
+            if item not in unique_items:
+                unique_items.append(item)
+                item_price = list(item.values())[0]
+                cost_of_unique_items = cost_of_unique_items + item_price
+        self.discount = round(self.get_total() - cost_of_unique_items, 2)
```

Comment:
Nice buy-one-get-one-free idea here! Although it does end up being buy-one-get-lots-free in certain circumstances 😄